inlanecruising_plugin/src/calculation.cpp

```diff
@@ -0,0 +1,308 @@
+    {
+      best_index = i;
+      min_distance = distance;
+    }
+    i++;
+  }
+
+  return best_index;
+}
+
+std::vector<double> moving_average_filter(const std::vector<double> input, int window_size) {
+  int i = 0;
+  double average;
+  std::deque<double> samples;
+  std::vector<double> output;
+  output.reserve(input.size());
+  for (auto value : input)
+  {
+
+    if (i < window_size) {
+      samples.push_back(value);
+    } else {
+      samples.pop_front();
+      samples.push_back(value);
+    }
+
+    double total = 0;
+    for (auto s: samples) {
+      total += s;
+    }
+
+    double average = total / samples.size();
+    output.push_back(average);
+    i++;
+  }
+
+  return output;
+}
+
+void splitPointSpeedPairs(const std::vector<PointSpeedPair>& points, std::vector<lanelet::BasicPoint2d>* basic_points,
+                          std::vector<double>* speeds)
+{
+  basic_points->reserve(points.size());
+  speeds->reserve(points.size());
+
+  for (const auto& p : points)
+  {
+    basic_points->push_back(std::get<0>(p));
+    speeds->push_back(std::get<1>(p));
+  }
+}
+
+std::vector<lanelet::BasicPoint2d> pointSpeedPairsToBasicPoints(const std::vector<PointSpeedPair>& points)
+{
+  std::vector<lanelet::BasicPoint2d> basic_points;
+  basic_points.reserve(points.size());
+
+  for (const auto& p : points)
+  {
+    basic_points.push_back(std::get<0>(p));
+  }
+
+  return basic_points;
+}
+
+boost::optional<tk::spline> compute_fit(std::vector<lanelet::BasicPoint2d> basic_points)
+{
+  if (basic_points.size() < 3)
+  {
+    ROS_WARN_STREAM("Insufficient Spline Points");
+    return boost::none;
+  }
+
+  tk::spline spl;
+  std::vector<double> points_x;
+  std::vector<double> points_y;
+
+  for (size_t i = 0; i < basic_points.size(); i++)
+  {
+    ROS_WARN_STREAM("basic_points[i]: " << basic_points[i].x() << ", " << basic_points[i].y());
+    points_x.push_back(basic_points[i].x());
+    points_y.push_back(basic_points[i].y());
+  }
+
+  spl.set_points(points_x, points_y);
+
+  return spl;
+}
+
+double calculate_yaw(std::vector<double> cur_point, std::vector<double> next_point)
+{
+  double dx = next_point[0] - cur_point[0];
+  double dy = next_point[1] - cur_point[1];
+  double yaw = atan2(dy, dx);
+  return yaw;
+}
+
+double calculate_curvature(std::vector<double> cur_point, std::vector<double> next_point)
+{
+  double dist = sqrt(pow(cur_point[0] - next_point[0], 2) + pow(cur_point[1] - next_point[0], 2));
+
+  double angle = calculate_yaw(cur_point, next_point);
+
+  double r = 0.5 * (dist / std::sin(angle));
+
+  double max_curvature = 100000;
+  double curvature = std::min(1 / r, max_curvature);
+
+  return curvature;
+}
+
+std::vector<double> compute_orientation_from_fit(tk::spline curve, std::vector<lanelet::BasicPoint2d> sampling_points)
+{
+  std::vector<double> orientations;
+  std::vector<double> cur_point{ 0.0, 0.0 };
+  std::vector<double> next_point{ 0.0, 0.0 };
+  double lookahead = 0.3;
```

Comment:
Unused variable?

Reply:
Left over from previous implementation. I'll take it out